bot/kodiak/test_pull_request.py

```diff
@@ -0,0 +1,238 @@
+from typing import Any, Type, cast
+
+import pytest
+import requests
+
+from kodiak.config import V1, Merge, MergeMethod
+from kodiak.errors import ApiCallException
+from kodiak.pull_request import PRV2, EventInfoResponse
+from kodiak.queries import (
+    BranchProtectionRule,
+    Client,
+    MergeableState,
+    MergeStateStatus,
+    PullRequest,
+    PullRequestAuthor,
+    PullRequestState,
+    RepoInfo,
+)
+
+
+def create_event() -> EventInfoResponse:
+    config = V1(
+        version=1, merge=Merge(automerge_label="automerge", method=MergeMethod.squash)
+    )
+    pr = PullRequest(
+        id="e14ff7599399478fb9dbc2dacb87da72",
+        number=100,
+        author=PullRequestAuthor(login="arnold", databaseId=49118, type="Bot"),
+        mergeStateStatus=MergeStateStatus.BEHIND,
+        state=PullRequestState.OPEN,
+        mergeable=MergeableState.MERGEABLE,
+        isCrossRepository=False,
+        labels=["automerge"],
+        latest_sha="8d728d017cac4f5ba37533debe65730abe65730a",
+        baseRefName="master",
+        headRefName="df825f90-9825-424c-a97e-733522027e4c",
+        title="Update README.md",
+        body="",
+        bodyText="",
+        bodyHTML="",
+        url="https://github.com/delos-corp/hive-mind/pull/324",
+    )
+    rep_info = RepoInfo(
+        merge_commit_allowed=False,
+        rebase_merge_allowed=False,
+        squash_merge_allowed=True,
+    )
+    branch_protection = BranchProtectionRule(
+        requiresApprovingReviews=True,
+        requiredApprovingReviewCount=2,
+        requiresStatusChecks=True,
+        requiredStatusCheckContexts=[
+            "ci/circleci: frontend_lint",
+            "ci/circleci: frontend_test",
+        ],
+        requiresStrictStatusChecks=True,
+        requiresCommitSignatures=False,
+    )
+
+    return EventInfoResponse(
+        config=config,
+        config_str="""\
+version = 1
+[merge]
+method = "squash"
+""",
+        config_file_expression="master:.kodiak.toml",
+        head_exists=True,
+        pull_request=pr,
+        repo=rep_info,
+        branch_protection=branch_protection,
+        review_requests=[],
+        reviews=[],
+        status_contexts=[],
+        check_runs=[],
+        valid_signature=True,
+        valid_merge_methods=[MergeMethod.squash],
+    )
+
+
+@pytest.fixture
+async def pr_v2() -> PRV2:
+    async def dequeue() -> None:
+        return None
+
+    async def queue_for_merge() -> None:
+        return None
+
+    return PRV2(
+        event=create_event(),
+        install="88443234",
+        owner="delos",
+        repo="incite",
+        number=8534,
+        dequeue_callback=dequeue,
+        queue_for_merge_callback=queue_for_merge,
+    )
+
+
+@pytest.mark.asyncio
+async def test_pr_v2_merge() -> None:
+    async def dequeue() -> None:
+        return None
+
+    async def queue_for_merge() -> None:
+        return None
+
+    class FakeClient:
```

Comment:
I think this could be moved to a function that takes in the byte string and returns a client